crates/astria-core/src/protocol/abci.rs

```diff
@@ -64,6 +65,9 @@ impl AbciErrorCode {
                 "the account has reached the maximum number of parked transactions".into()
             }
             Self::PARKED_FULL => "the mempool is out of space for more parked transactions".into(),
+            Self::NONCE_REPLACEMENT => {
+                "the transaction was replaced by a different transaction with the same nonce".into()
```

Comment:
What will the flow look like to actually see this abci error code?

If I understand the follow correctly:

1. transaction is stuck in mempool
2. new transaction is submitted to the mempool, replacing the old one
3. check-tx for the new transaction should return a success code (that is abci code 0).

But `NONCE_REPLACEMENT` is an error code for a transaction that got replaced if I understand correctly.

Reply:
The next task I have is for general transaction observability (#1773), we don't currently have the infra to communicate it now to the users but we should

Reply:
I don't think one this is strictly related to the other. The `AbciErrorCode`s are intended to be returned in response to abci queries and put into the `code` field of the various responses (in this case `CheckTx`).

All the ABCI error codes can be observed right now (even if clunky). I don't understand when/how this code would be observed.